sig-robotics/proposal/Generating Action Sequences for Robot Task Plans using LLM/README.md

```diff
@@ -0,0 +1,176 @@
+
+1. Test the perception, planning, and control algorithms of a robot in complex navigation and grasping tasks, both indoors and outdoors, based on given instructions (e.g. "Go to the kitchen and help me get an apple."). Utilize large language models like gpt-3.5-turbo for this purpose.
+
+2. Perform closed-loop verification of the developed algorithms in **[Gazebo](https://gazebosim.org/home)** simulation environment.
+
+3. Deploy the data transmission model and other development processes using RoboSDK.
+
+
+# 3. Proposals
+By building RoboAgent and its related components on RoboSDK, RoboSDK can have the ability to handle complex robot tasks based on LLM (Large Language Models).
+
+The scope of RoboAgent and its related suite includes:
+- Building a RoboAgent capable of task planning, environment observation, and phenomena reasoning based on prompt techniques such as ReAct, self-ask, tree-of-thoughts, to enable complex task planning and processing abilities.
+- Providing the ability to utilize large language models for suites such as RoboAgent and RoboToolKit.
+- Building the RoboToolKit suite based on prompt techniques like ReAct, which interfaces with the control interface built on RoboSDK to provide action-related instructions and control capabilities to RoboAgent.
+- Providing the ability for RoboAgent to perceive the environment by accessing the sensor interface data of RoboSDK.
+- Providing relevant documentation, sample templates, and auxiliary tools to reduce the learning curve for users.
+- Creating related test cases and test reports.
+
+This design targets developers aiming to build complex robots based on LLM using RoboSDK.
+
+# 4. Design Details
+
+## Architecture Design
+To better build the capabilities of RoboAgent in task planning, environment perception, and reasoning, and to enable RoboAgent to handle complex robot instructions, we need to build the RoboAgent and related tool suites, including Planning, Action, Reasoning, and Observation. In addition, we need to address issues regarding the illusion of large language models and the effectiveness of result outputs. It is necessary to incorporate validators and evaluation components into the system architecture for proper assessment. The following diagram illustrates the system architecture built based on this concept.
+
+![System Architecture](./images/image_3.png)
+
+
+
+In this architecture:
+- Users send commands to ROS robots through KubeEdge edge nodes. KubeEdge provides the communication and management infrastructure across edge and cloud.
+- In ROS, a node or module is created to communicate with KubeEdge, which integrates directly with the existing RoboAgent for communication.
+- After receiving user requests from KubeEdge, the RoboAgent uses LLM for task planning and generates a sequence of subtasks. Task planning is based on user input and robot capabilities, determining the optimal way to accomplish the required tasks.
```

Comment:
In "The RoboAgent uses LLM", I can not find the LLM module in your architechture. So I suggest a new module named "LLM" should be placed in the architechture, and the message bewteen RoboAgent and LLM should also be clearly pointed out.